lakey_service/downloader/executors/baseexecutor.py

```diff
@@ -0,0 +1,56 @@
+
+class BaseExecutor:
+
+    def execute(self, download_request):
+        raise NotImplementedError()
+
+    def get_sample(self, catalogue_item):
+        raise NotImplementedError()
+
+    def get_size(self, column_name, catalogue_item):
+        raise NotImplementedError()
+
+    def get_distribution(
+            self, column_name, column_type, column_is_enum, catalogue_item):
+        raise NotImplementedError()
+
+    def compile_to_query(self, download_request):
+
+        table = download_request.catalogue_item.name
+        spec = download_request.spec
+
+        # -- COLUMNS
+        columns = ', '.join(spec['columns'])
+
+        # -- FILTERS
+        filters_entries = []
+        for fltr in spec['filters']:
```

Comment:
for now this is a poor's man approach to string building. If this would start to get more complex we could switch to `render_to_string` (https://docs.djangoproject.com/en/3.0/topics/templates/#django.template.loader.render_to_string) and use django.s powerful template engine.